kie-pmml-new/kie-pmml-models/kie-pmml-models-tree/kie-pmml-models-tree-compiler/src/main/java/org/kie/pmml/models/tree/compiler/factories/KiePMMLASTFactoryUtils.java

```
@@ -0,0 +1,135 @@
+
+import static java.util.stream.Collectors.groupingBy;
+import static org.kie.pmml.commons.utils.DrooledModelUtils.getCorrectlyFormattedResult;
+
+public class KiePMMLASTFactoryUtils {
+
+    private KiePMMLASTFactoryUtils() {
+        // not instatiate
+    }
+
+    public static List<KiePMMLFieldOperatorValue> getConstraintEntriesFromAndOrCompoundPredicate(final CompoundPredicate compoundPredicate, final Map<String, KiePMMLOriginalTypeGeneratedType> fieldTypeMap) {
+        // Managing only SimplePredicates for the moment being
+        final List<Predicate> simplePredicates = compoundPredicate.getPredicates().stream().filter(predicate -> predicate instanceof SimplePredicate).collect(Collectors.toList());
+        if (!CompoundPredicate.BooleanOperator.AND.equals(compoundPredicate.getBooleanOperator()) &&
+                !CompoundPredicate.BooleanOperator.OR.equals((compoundPredicate.getBooleanOperator()))) {
+            throw new KiePMMLException("getConstraintEntriesFromAndOrCompoundPredicate invoked with a " + compoundPredicate.getBooleanOperator() + " CompoundPredicate");
+        }
+        final Map<String, List<SimplePredicate>> predicatesByField = simplePredicates.stream()
+                .map(child -> (SimplePredicate) child)
+                .collect(groupingBy(child -> fieldTypeMap.get(child.getField().getValue()).getGeneratedType()));
+        final List<KiePMMLFieldOperatorValue> toReturn = new LinkedList<>();
+        switch (compoundPredicate.getBooleanOperator()) {
+            case AND:
+                predicatesByField.forEach((fieldName, predicates) -> toReturn.add(getConstraintEntryFromSimplePredicates(fieldName, "&&", predicates, fieldTypeMap)));
+                break;
+            case OR:
+                predicatesByField.forEach((fieldName, predicates) -> toReturn.add(getConstraintEntryFromSimplePredicates(fieldName, "||", predicates, fieldTypeMap)));
+                break;
+            default:
+                break;
```

Please raise an exception like `new IllegalStateException("This should never happen")`

---

Done